sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +64,27 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Unit = {
```

Comment:
let's add some comments to explain the motivation: `source.Filter` can't cover all the expressions, and we need this hack to avoid perf regression in file source v2.